File: facade/typescript/Facade.ts
// Subsystem: CPU
class CPU {
  freeze(): string {
    return 'CPU: Freezing processor';
  }

  jump(position: string): string {
    return `CPU: Jumping to position ${position}`;
  }

  execute(): string {
    return 'CPU: Executing instructions';
  }
}

// Subsystem: Memory
class Memory {
  load(position: string, data: string): string {
    return `Memory: Loading '${data}' at position ${position}`;
  }
}

// Subsystem: HardDrive
class HardDrive {
  read(sector: string, size: string): string {
    return `HardDrive: Reading ${size} bytes from sector ${sector}`;
  }
}

// Facade
class ComputerFacade {
  private cpu: CPU;
  private memory: Memory;
  private hardDrive: HardDrive;

  constructor() {
    this.cpu = new CPU();
    this.memory = new Memory();
    this.hardDrive = new HardDrive();
  }

  start(): string[] {
    const results: string[] = [];
    results.push('Starting Computer...');
    results.push(this.cpu.freeze());
    results.push(this.memory.load('0x00', 'BOOT_ADDRESS'));
    results.push(this.hardDrive.read('0', '1024'));
    results.push(this.cpu.jump('0x00'));
    results.push(this.cpu.execute());
    results.push('Computer started successfully!');
    return results;
  }
}

// Usage
const computer = new ComputerFacade();
const results = computer.start();
results.forEach(result => console.log(result));

File: facade/java/Facade.java
import java.util.ArrayList;
import java.util.List;

// Subsystem: CPU
class CPU {
    public String freeze() {
        return "CPU: Freezing processor";
    }

    public String jump(String position) {
        return "CPU: Jumping to position " + position;
    }

    public String execute() {
        return "CPU: Executing instructions";
    }
}

// Subsystem: Memory
class Memory {
    public String load(String position, String data) {
        return "Memory: Loading '" + data + "' at position " + position;
    }
}

// Subsystem: HardDrive
class HardDrive {
    public String read(String sector, String size) {
        return "HardDrive: Reading " + size + " bytes from sector " + sector;
    }
}

// Facade
class ComputerFacade {
    private CPU cpu;
    private Memory memory;
    private HardDrive hardDrive;

    public ComputerFacade() {
        this.cpu = new CPU();
        this.memory = new Memory();
        this.hardDrive = new HardDrive();
    }

    public List<String> start() {
        List<String> results = new ArrayList<>();
        results.add("Starting Computer...");
        results.add(cpu.freeze());
        results.add(memory.load("0x00", "BOOT_ADDRESS"));
        results.add(hardDrive.read("0", "1024"));
        results.add(cpu.jump("0x00"));
        results.add(cpu.execute());
        results.add("Computer started successfully!");
        return results;
    }
}

// Main class demonstrating Facade pattern
public class Facade {
    public static void main(String[] args) {
        ComputerFacade computer = new ComputerFacade();
        List<String> results = computer.start();
        for (String result : results) {
            System.out.println(result);
        }
    }
}
